frontend/src/components/layout/ProfileButton.tsx

```diff
@@ -7,10 +7,11 @@ import useOutsideClick from "@/hooks/useOutsideClick";
 function ProfileButton() {
   const [open, setOpen] = useState(false);
   const user = useStore((state) => state.user);
+  const store = useStore.getState();
```

Comment:
Please use the store like this. This will ensure that this component is only rerendered when this specific key changes, not the hole store.

```suggestion
  const { isAuthenticated } = useStore((state) => ({
    isAuthenticated: state.auth.isAuthenticated,
  }));
```